### ISSUE TITLE
Bring sync api back for a few minor verions

### ISSUE THREAD

**Issue by fa93hws** [testing-library/user-event#1177]
### Problem description

One breaking change introduced in v14 is that almost all api is changed to `async` now.
It become a nightmare to upgrade testing library now because we lots of frontend engineer works on the monorepo and it's extremely hard to get all usage changed in one go.

I think for a breaking api that has such a big impact maybe better to have a smoother migration plan?

### Suggested solution

Bring the sync api back for a few minor version and decorated as `deprecated`. They can be removed from next major version.
So that we can upgrade first, enforce a linting rule and fix the rest progressively.

**Comment by ethanneff** [comment]
yeah we are still on v13 because v14 was a nightmare to migrate

**Comment by eddielee394** [comment]
Yeah, i"m in the process of doing the upgrade now on a medium sized application and it's been brutal.  We have 3 other FE apps that we still have to do that I'm genuinely not looking forward to.

**Comment by ph-fritsche** [comment]
We will neither have the capacity to write nor to maintain two implementations and moving to asynchronous implementations was necessary for some features to be implemented.

If you use a [`setup function`](https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent) that you share across all tests, you can easily replace most calls with find&replace of your IDE:
```js
test('title', () => // becomes
test('title', async () => //...

render(// becomes
const {user} = setup(//...

userEvent.// becomes
await user.// ...
```
If your tests rely on asynchronous user operations to happen in one step of the event loop, you can set `delay: null` in your setup function and override it in one suite at a time.
